django/cantusdb_project/main_app/tests/test_views.py

```diff
@@ -871,6 +871,487 @@ def test_keyword_search_contains(self):
         context_chant_id = response.context["chants"][0]["id"]
         self.assertEqual(chant.id, context_chant_id)
 
+    def test_order_by_siglum(self):
+        source_1 = make_fake_source(published=True, siglum="sigl-1")
+        chant_1 = make_fake_chant(incipit="thing 1", source=source_1)
+        source_2 = make_fake_source(published=True, siglum="sigl-2")
+        chant_2 = make_fake_chant(incipit="thing 2", source=source_2)
+
+        search_term = "thing"
+
+        response_ascending = self.client.get(
+            reverse("chant-search"),
+            {
+                "keyword": search_term,
+                "op": "contains",
+                "order": "siglum",
+                "sort": "asc",
+            },
+        )
+        ascending_results = response_ascending.context["chants"]
+        first_result_incipit = ascending_results[0]["incipit"]
+        self.assertEqual(first_result_incipit, chant_1.incipit)
+        last_result_incipit = ascending_results[1]["incipit"]
+        self.assertEqual(last_result_incipit, chant_2.incipit)
+
+        response_descending = self.client.get(
+            reverse("chant-search"),
+            {
+                "keyword": search_term,
+                "op": "contains",
+                "order": "siglum",
+                "sort": "desc",
+            },
+        )
+        descending_results = response_descending.context["chants"]
+        first_result_incipit = descending_results[1]["incipit"]
+        self.assertEqual(first_result_incipit, chant_1.incipit)
+        last_result_incipit = descending_results[0]["incipit"]
+        self.assertEqual(last_result_incipit, chant_2.incipit)
```

Comment:
I think it would be clearer if here (and in the other ordering tests) you still set the key in `first_result_incipit = descending_results[1]["incipit"]` to 0 instead of 1, otherwise the "first result" variable is set to the second (last) result. Then in the following line say `self.assertEqual(first_result_incipit, chant_2.incipit)`

When I read through the first time it looked like there wasn't any difference from the first block.

Reply:
Agreed. `first_result_incipit` and `last_result_incipit` variable names are confusing here.

Reply:
this was an oversight on my part - apologies, and thanks for catching this.